src/hooks/useCreatePoolHandler.tsx

```diff
@@ -1 +1,88 @@
-export const useCreatePoolHandler = () => {}
+import { useContext, useEffect, useState } from 'react'
+import { ethers, Contract } from 'ethers'
+import { WalletContext } from '../../src/context/WalletContext'
+import GammaPoolFactory from '../../abis/v1-core/GammaPoolFactory.sol/GammaPoolFactory.json'
+import toast from 'react-hot-toast'
+import { calcPoolKey } from '../../src/utils/getSmartContract'
+
+export const useCreatePoolHandler = () => {
+
+  const [gammaPoolFactory, setGammaPoolFactory] = useState<Contract | null>(null)
+  const [token1Addr, setToken1Addr] = useState<string>('')
+  const [token2Addr, setToken2Addr] = useState<string>('')
+  const [cfmmAddr, setCfmmAddr] = useState<string>('')
+
+  const { accountInfo, provider, signer } = useContext(WalletContext)
+  useEffect(() => {
+    async function fetchContract() {
+      if (!provider) {
+        console.log('Please connect wallet.')
+        return
+      }
+
+      let gammaFactoryaddr = process.env.NEXT_PUBLIC_GAMMAFACTORY_ADDR || ''
+      let _gammaPoolFactory = new ethers.Contract(
+        gammaFactoryaddr,
+        GammaPoolFactory.abi,
+        accountInfo && accountInfo?.address ? signer : provider
+      )
+
+      if (_gammaPoolFactory) {
+        setGammaPoolFactory(_gammaPoolFactory)
+      }
+    }
+    fetchContract()
+  }, [provider])
+
+  const createPool = async () => {
+    try {
+      if (!accountInfo || !gammaPoolFactory) {
+        toast.error("Please connect your wallet.")
+        return
+      }
+
+      // check if pool already exists
+      let pool = await gammaPoolFactory.getPool(calcPoolKey(cfmmAddr))
+      if (pool != ethers.constants.AddressZero) {
+        toast.error("Pool already exsts at " + pool)
+        return
+      }
+
+      let CreatePoolParams = {
+        cfmm: cfmmAddr,
+        tokens: [token1Addr, token2Addr],
+        protocol: 1,
+      }
+      let tx = await gammaPoolFactory.createPool(CreatePoolParams)
+      let res = await tx.wait()
+      let poolAddress = res.events[0].address
+      let msg = 'Pool created successfully at address: ' + poolAddress
+      toast.success(msg)
+      console.log(msg)
```

Comment:
Why are we using `toast.success()` instead of `notifyError()`? We should just adopt one of the two.

Reply:
NotifyError/ Success doesn't seem to be working.

Reply:
https://github.com/gammaswap/v1-interface/blob/e9d0cd4162db9c9f893f751aad33ef1436bd4f4e/src/hooks/useNotification.tsx#L1-L12
I believe this is happening because you need to import `react-hot-toast` in here and define the function here. This is so that you wouldn't have to import it every time you're going to use it. I think we should keep `react-hot-toast`, as long as we are able to customize the styling later.

Reply:
added react-hot-toast to useNotification